### PR TITLE
feat: add wait for transaction receipt hook

### PR DESCRIPTION
**Pull request by abcrane123**
**What changed? Why?**
- wait for transaction receipt before labeling successful

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/transaction/types.ts [hunk 1]
```diff
@@ -21,10 +25,12 @@ export type TransactionContextType = {
   isLoading: boolean; // A boolean indicating if the transaction is currently loading.
   isToastVisible: boolean; // A boolean indicating if the transaction toast notification is visible.
   onSubmit: () => void; // A function called when the transaction is submitted.
+  receipt?: TransactionReceipt;
```
Comment (Zizzamia) [inline src/transaction/types.ts:28]: Oh niiiice that we reuse the same type

File: src/transaction/hooks/useGetTransactionStatus.tsx [hunk 2]
```diff
@@ -68,8 +70,10 @@ export function useGetTransactionStatus() {
     accountChainId,
     errorMessage,
     isLoading,
+    receipt,
```
Comment (Zizzamia) [inline src/transaction/hooks/useGetTransactionStatus.tsx:73]: `receipt` is after `onSubmit` ;)

File: src/transaction/types.ts (hunk 1 above)
Comment (cpcramer) [reply src/transaction/types.ts:28]: Nit - Add comment explaining `receipt`

File: src/transaction/utils.ts [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
   transactionHash,
   transactionId,
 }: IsSpinnerDisplayedProps) {
-  if (transactionHash) {
+  if (hasReceipt) {
     return false;
   }
   if (errorMessage) {
     return false;
   }
```
Comment (cpcramer) [inline src/transaction/utils.ts:25]:
```suggestion
  if (hasReceipt || errorMessage) {
    return false;
  }
```

File: src/transaction/components/TransactionButton.tsx [hunk 4]
```diff
@@ -14,21 +16,39 @@ export function TransactionButton({
...
+    (isLoading ||
+      !contracts ||
+      !address ||
+      transactionId ||
+      statusWriteContract === 'pending' ||
+      statusWriteContracts === 'pending');
 
   const displaySpinner = isSpinnerDisplayed({
     errorMessage,
+    hasReceipt: !!receipt,
```
Comment (cpcramer) [inline src/transaction/components/TransactionButton.tsx:37]: Is there an easier way we can write this? Double negatives are confusing haha

Comment (cpcramer) [reply src/transaction/components/TransactionButton.tsx:37]: Oh I see what it's doing

File: src/transaction/components/TransactionButton.tsx [hunk 5]
```diff
@@ -14,21 +16,40 @@ export function TransactionButton({
...
     transactionId,
   } = useTransactionContext();
 
-  const isDisabled = isLoading || !contracts || !address || transactionId;
+  const isInProgress =
+    statusWriteContract === 'pending' ||
+    statusWriteContracts === 'pending' ||
+    isLoading;
+  const isMissingProps = !contracts || !address;
+  const isWaitingForReceipt = transactionId || transactionHash;
```
Comment (Zizzamia) [inline src/transaction/components/TransactionButton.tsx:31]: Love the clarity of those states.

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**